app/implementations/openssl/3/iut.c

```diff
@@ -103,11 +103,14 @@ ACVP_RESULT iut_register_capabilities(ACVP_CTX *ctx, APP_CONFIG *cfg) {
     if (!fips_ver) {
         return register_capabilities_non_fips(ctx, cfg);
     } else if (fips_ver <= OPENSSL_FIPS_309) {
+        printf("Registering capabilities for FP 3.0.X...\n");
         /* 3.0.X registrations are so nearly identical, we left them combined. */
         return register_capabilities_fp_30X(ctx, cfg);
-    } else if (fips_ver >= OPENSSL_FIPS_312) {
+    } else if (fips_ver <= OPENSSL_FIPS_312) {
```

Comment:
so if ver is less than 312 you still register for 312 ?

Reply:
if ver is between 3.0.9 and 3.1.2, yes - those FPs are not certified so we don't really support that, though nothing is preventing the tests from running and seeing what happens.